pkg/deploy/dev-workspace-config/dev_workspace_config.go

```diff
@@ -104,6 +105,13 @@ func updateWorkspaceConfig(cheCluster *chev2.CheCluster, operatorConfig *control
 
 	updateWorkspaceImagePullPolicy(devEnvironments.ImagePullPolicy, operatorConfig.Workspace)
 
+	if ctx.Proxy.HttpProxy != "" || ctx.Proxy.HttpsProxy != "" {
```

Comment:
1. Could you add some comments explaining why we have to clean up proxy config, please?
2. There is something that I don't understand. If proxy is set, then we have to clean up to DWO proxy config. If proxy is not set, then we do nothing. What is the point of DWO proxy config then?

Reply:
Yup, I agree some comments here would be helpful.

As for why DWO supports configuring the proxy, the answer is twofold: first, it exists as an option because we implemented it (way back in v0.12, before proxy was supported in Che Operator); second, it's a feature that's still used by other consumers of DevWorkspaces, such as Web Terminal.

Che introduced its own implementation of managing proxy settings for DevWorkspaces, but the two implementations collide -- Che supports the proxy by creating a `che-proxy-settings` configmap which is mounted as env vars, and so needs to disable built-in DWO proxy configuration/detection. 

Basically, what we're doing here is telling DWO "we'll handle setting up proxy environment variables, you shouldn't do it for this workspace".